Fixes ip6tables errors on RHEL 6/7

wilson208 opened the pull request:
Caused through merge of PR's #658 and then subsequently this older PR #641

DavidS commented on manifests/linux/redhat.pp:
```diff
@@ -62,58 +62,62 @@
+      }
+    }
+    default:     {}
+    #lint:endignore
```
The other changes seem fine, but why does this block have to move?

https://github.com/puppetlabs/puppetlabs-firewall/blob/master/lib/puppet/type/firewall.rb#L1531 will also require updating.

wilson208 commented:
@DavidS I forgot to run spec and lint on this the first time around (oops), I have updated this to make spec tests pass and fix lint errors.

Have addressed your comment on moving that case block, have reverted from using $seluser and $seltype variables to keep order the same. Have also updated the firewall type.
